### PR TITLE
[Plinth] [Builtins] Remove 'CaseList' and 'CaseData'

### PR DESCRIPTION
**Pull request by effectfully**
This removes usages of `CaseList` and `CaseData` from Plinth by compiling those to `ChooseList`- and `ChooseData`-based functions. This is just so that we can test the constitution script, the intention is to revert this PR once `CaseList` and `CaseData` are costed.

### CODE REVIEW

**Review by zliu41 — COMMENTED** [review]
> This is just so that we can test the constitution script, the intention is to revert this PR once CaseList and CaseData are costed.

We can't revert this until `caseList` etc. are enabled on mainnet.

File: plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs [hunk 1]
```diff
@@ -409,7 +410,49 @@ defineBuiltinTerms = do
...
+                -- > /\a r ->
+                -- >   \(z : r) (f : a -> list a -> r) (xs : list a) ->
+                -- >     chooseList
+                -- >       {a}
+                -- >       {all dead. r}
+                -- >       xs
+                -- >       (/\dead -> z)
+                -- >       (/\dead -> f (headList {a} xs) (tailList {a} xs))
+                -- >       {r}
+                fmap (const annMayInline) . runQuote $ do
```
Comment (zliu41) [inline plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs:423]: Why not just revert the changes to `PlutusTx.Builtins` and `PlutusTx.Builtins.Internal`? This is more complex and harder to follow.

File: plutus-tx-plugin/test/IsData/9.6/deconstructData.pir.golden
Comment (effectfully) [inline plutus-tx-plugin/test/IsData/9.6/deconstructData.pir.golden:1]: I took the comments from this golden file.

File: plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs (hunk 1 above)
Comment (effectfully) [reply plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs:423]: BTW, note that I didn't just write it down from scratch like a maniac, I copy-pasted definitions from `PlutusCore.StdLib.Data.List` and `PlutusCore.StdLib.Data.Data` and tweaked them just a little to get the right argument ordering and `all dead. <...>` instead of `() -> <...>`.

**Review by zliu41 — APPROVED** [review]
